optimade/adapters/structures/__init__.py

```diff
@@ -0,0 +1,89 @@
+# pylint: disable=undefined-variable
+from typing import Union
+
+from optimade.models import StructureResource
+
+from .aiida import get_aiida_structure_data
+from .ase import get_ase_atoms
+from .cif import get_cif
+from .proteindatabank import get_pdb, get_pdbx_mmcif
+from .pymatgen import get_pymatgen_structure
+
+
+__all__ = ("Structure",)
+
+
+class Structure:
+    """Lazy structure converter
+    :param structure: JSON OPTIMADE single structures resource entry.
+    """
+
+    _structure_type_converters = {
+        "aiida_structuredata": get_aiida_structure_data,
+        "ase": get_ase_atoms,
+        "cif": get_cif,
+        "pdb": get_pdb,
+        "pdbx_mmcif": get_pdbx_mmcif,
+        "pymatgen": get_pymatgen_structure,
+    }
```

Comment:
Eventually, this might become a `ConverterCollection` class that could include checks on the interface of the functions it contains etc.
For the moment, this is fine

Reply:
Great idea. And definitely one for the future, i.e., not something I will explore at this time.